### ISSUE TITLE
[Injury Prediction Phase 3] ML Model Training & Evaluation

### ISSUE THREAD

**Issue by jpequegn** [jpequegn/ai-coach#50]
**Parent Issue:** #30 - Injury Prediction Model
**Depends On:** #49 (Phase 2)

## Scope
Train machine learning models to predict injury risk, evaluate performance, and implement model explainability.

### Training Data Preparation
- [ ] Collect historical training data with features
- [ ] Label injury events (binary or risk score)
- [ ] Handle class imbalance:
  - Implement SMOTE (Synthetic Minority Oversampling)
  - Add class weighting
  - Use stratified sampling
- [ ] Split data: 70% train, 15% validation, 15% test
- [ ] Implement data augmentation strategies

### Model Training
- [ ] Create `InjuryPredictionTrainer` service
- [ ] Train Random Forest classifier/regressor
- [ ] Train Gradient Boosting model
- [ ] Train Logistic Regression (baseline)
- [ ] Implement hyperparameter tuning with cross-validation
- [ ] Perform feature importance analysis
- [ ] Save trained models with versioning

### Model Evaluation
- [ ] Evaluate on test set:
  - Precision, Recall, F1-score
  - ROC-AUC curve
  - Precision-Recall curve
  - Confusion matrix
- [ ] Compare model performance
- [ ] Analyze false positives and false negatives
- [ ] Select best-performing model
- [ ] Document model performance metrics

### Model Explainability
- [ ] Implement feature importance extraction
- [ ] Generate per-prediction explanations
- [ ] Create risk factor breakdown
- [ ] Add confidence scores to predictions
- [ ] Document decision logic

### Model Persistence
- [ ] Implement model serialization with `serde`
- [ ] Create model versioning system
- [ ] Store models in database or filesystem
- [ ] Implement model loading on service initialization
- [ ] Add model update mechanism

## Deliverables
- Trained ML models (Random Forest, GBM, Logistic Regression)
- Model evaluation report
- Model explainability system
- Model versioning and persistence

## Dependencies
- Phase 2: Feature engineering complete
- Historical training data with injury labels
- `linfa` ML framework

## Estimated Time
3 weeks

## Testing
```bash
# Train models
cargo run --bin train_injury_model

# Evaluate models
cargo test injury_model_evaluation

# Check model performance
cat models/injury_prediction_v1_metrics.json
```

## Acceptance Criteria
- [ ] All three models trained successfully
- [ ] Test AUC-ROC >0.70
- [ ] Best model selected based on metrics
- [ ] Feature importance documented
- [ ] Models serialized and loadable
- [ ] Confidence scores generated
